src/overlays/actors/ovl_En_Talk_Gibud/z_en_talk_gibud.c

```diff
@@ -28,166 +83,1065 @@ const ActorInit En_Talk_Gibud_InitVars = {
+        if (this->effectTimer > 0 && this->effectType == 0 && this->type == EN_TALK_GIBUD_TYPE_GIBDO) {
+            this->actor.hintId = 0x2A;
+            this->actor.flags &= ~(0x8 | 0x1);
+            this->actor.flags |= (0x4 | 0x1);
+            SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06010B88, NULL, this->jointTable, this->morphTable, 26);
+            this->type = EN_TALK_GIBUD_TYPE_REDEAD;
+        }
+        if (EnTalkGibud_PlayerOutOfRange(this, globalCtx)) {
+            EnTalkGibud_SetupWalkToHome(this);
+        } else {
+            EnTalkGibud_SetupWalkToPlayer(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk_Gibud/func_80AFF330.s")
+void EnTalkGibud_SetupDead(EnTalkGibud* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, EN_TALK_GIBUD_ANIMATION_DEATH);
+    this->actor.flags &= -2;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_DEAD);
+    this->deathTimer = 0;
+    this->actionFunc = EnTalkGibud_Dead;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk_Gibud/func_80AFF378.s")
+void EnTalkGibud_Dead(EnTalkGibud* this, GlobalContext* globalCtx) {
+    if (this->deathTimer > 300) {
+        EnTalkGibud_SetupRevive(this);
+    } else {
+        Math_SmoothStepToS(&this->headRotation.y, 0, 1, 250, 0);
+        Math_SmoothStepToS(&this->upperBodyRotation.y, 0, 1, 250, 0);
+        this->deathTimer++;
+    }
+    if (this->deathTimer == 20 && this->effectTimer > 0 && this->effectType == 0 &&
+        this->type == EN_TALK_GIBUD_TYPE_GIBDO) {
+        SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06010B88, NULL, this->jointTable, this->morphTable, 26);
+        this->type = EN_TALK_GIBUD_TYPE_REDEAD;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk_Gibud/func_80AFF45C.s")
+void EnTalkGibud_SetupRevive(EnTalkGibud* this) {
+    Animation_Change(&this->skelAnime, &D_06009298, -1.0f, Animation_GetLastFrame(&D_06009298), 0.0f, 2, -8.0f);
+    this->actor.flags |= 1;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_REVERSE);
+    this->deathTimer = 0;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actionFunc = EnTalkGibud_Revive;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk_Gibud/func_80AFF4AC.s")
+void EnTalkGibud_Revive(EnTalkGibud* this, GlobalContext* globalCtx) {
+    if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+        this->actor.colChkInfo.health = 8;
+        EnTalkGibud_SetupIdle(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk_Gibud/func_80AFF618.s")
+void EnTalkGibud_GetTextIdForRequestedItem(EnTalkGibud* this, GlobalContext* globalCtx) {
+    switch (this->requestedItemIndex) {
+        case 0:
+            // Leeeave me something blue that bestows health!
+            func_801518B0(globalCtx, 0x138C, &this->actor);
+            this->textId = 0x138C;
+            break;
```

Comment:
Friendly reminder that we don't use actual quotes from the game in actors like this. You can put some small text that is similar but not exactly the same.
I haven't checked if this is the case, but just to be sure.